---
title: "Transcriptome sequencing of whole cell and mito fractions of sedentary and trained rats"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
    rmarkdown::html_document:
        toc: true
        toc_float: true
theme: cosmo
---

## Methods

Rats were kept in sedentary conditions or were trained. 
RNA was isolated from whole cell and mito fractions.
Libraries were generated by the Deakin Genomics Facility and sequenced on the NovaSeq system.

Fastqc and MultiQC were run to summarise the QC checks that were done.

Reads were then were mapped to the rat genome (Ensembl version 99) with Kallisto then
imported to R for analysis with DESeq2.
Pathway level analysis was then done using mitch with Reactome gene sets.

## Read counts

Import the Kallisto transcript counts.
We can also include some info out of the Ensembl GTF file including gene name
and gene class.

```{r,begin}

# libraries
library("reshape2")
library("DESeq2")
library("mitch")
library("gplots")
library("kableExtra")
library("beeswarm")

# import the 3 column table
tmp<-read.table("3col.tsv.gz",header=F)

# convert the 3 col table into a standard count matrix 
x<-as.matrix(acast(tmp, V2~V1, value.var="V3"))
# tidy up the column headers
colnames(x)<-sapply(strsplit(colnames(x),"_"),"[[",1)
head(x)

#dont forget gene names
g<-read.table("../ref2/Rattus_norvegicus.mRatBN7.2.106.gnames.txt",row.names=1)
g$gene_ID <- paste(g$V2,g$V3,g$V4,g$V5)
head(g)
g <- g[,ncol(g),drop=FALSE]
x<-merge(g,x,by=0)
rownames(x) <- x[,1]
x[,1]=NULL
# aggregate Tx data to genes
xx <- aggregate(. ~ gene_ID,x,sum)
# now round to integers so that DESeq2 doesn't fail
rownames(xx) <- xx[,1]
xx[,1]=NULL
x <- round(xx)
head(x)

write.table(x,file="countmatrix_skeletal.tsv",quote=FALSE,sep="\t")

y <- x/colSums(x)*1000000

write.table(y,file="rpmmatrix_skeletal.tsv",quote=FALSE,sep="\t")


```

## Samplesheet

```{r,samplesheet}

samplesheet <- read.table("samplesheet.tsv",header=TRUE)
samplesheet$UDI <- gsub("UDI_0","",samplesheet$UDI)
samplesheet$UDI <- gsub("UDI_","",samplesheet$UDI)
samplesheet$label <- paste(samplesheet$fraction,samplesheet$Group)
samplesheet <- samplesheet[order(samplesheet$UDI),]
samplesheet %>% kbl() %>% kable_paper("hover", full_width = F)

```

## Overall clustering with multidimensional scaling

Sample 95 is a HUMAN mito control sample.
Sample 96 is a RAT mito control sample.
These should be excluded from main results.

```{r,mds1,fig.height=8,fig.width=8}

ss <- samplesheet
ss <- ss[order(ss$UDI),]
colours = c('pink', 'lightblue','lightgreen','gray')
mds <- cmdscale(dist(t(x)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds*1.05 , cex=2, pch=19, xlab="Coordinate 1", ylab="Coordinate 2",
  col = colours[as.factor(ss$label)] ,  type = "p" ,
  xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(x) )
legend('topright', col=colours, legend=levels(as.factor(ss$label)), pch = 16, cex = 1)

x <- x[,which(! colnames(x) %in% c("95","96"))]

colours = c('pink', 'lightblue','lightgreen','gray')
mds <- cmdscale(dist(t(x)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds*1.05 , cex=2, pch=19, xlab="Coordinate 1", ylab="Coordinate 2",
  col = colours[as.factor(ss$label)] ,  type = "p" ,
  xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(x) )
legend('topright', col=colours, legend=levels(as.factor(ss$label)), pch = 16, cex = 1)


```

```{r,nreads,fig.height=8,fig.width=8}

ss$nreads<-colSums(x)
par(mar=c(5,10,5,3))
barplot(colSums(x),horiz=TRUE,las=2,main="number of reads per sample",cex.names=0.5)
par(mai=c(1.02,0.82,0.82,0.42))

```

## Check purity of mito fraction samples

Here I'm quantifying the mitochondrial read fraction.
That is the number of mt reads divided by the total number of reads.
We can see that purity is highly variable.

```{r,mito_purity,fig.height=8,fig.width=8}

par(mar=c(5,10,5,3))
mtfrac <-  colSums(x[grep("^Mt",rownames(x)),]) / colSums(x) 
barplot(mtfrac,horiz=TRUE,las=2,main="Proportion mitochondrial reads",cex.names=1)
par(mai=c(1.02,0.82,0.82,0.42))

mylevels <- levels(as.factor(ss$label))
mylevels
y <- x[,which(ss$label==mylevels[1])]
mitoS <- colSums(y[grep("^Mt",rownames(y)),]) / colSums(y)
y <- x[,which(ss$label==mylevels[2])] 
mitoT <- colSums(y[grep("^Mt",rownames(y)),]) / colSums(y)
y <- x[,which(ss$label==mylevels[3])] 
wholeS <- colSums(y[grep("^Mt",rownames(y)),]) / colSums(y)
y <- x[,which(ss$label==mylevels[4])] 
wholeT <- colSums(y[grep("^Mt",rownames(y)),]) / colSums(y)

dat <- list(mitoS,mitoT,wholeS,wholeT)
boxplot(dat,names=mylevels,ylab="mito frac")

median(wholeS)
median(wholeT)

table(mitoS>0.1)
table(mitoT>0.1)

ss$mtfrac <- mtfrac

ss %>% kbl() %>% kable_paper("hover", full_width = F)

mito <- subset(ss,fraction=="mito")

plot(mito$nreads,mito$mtfrac,pch=19,col="lightblue",xlab="No. reads", ylab="mtDNA proportion",cex=3)
text(mito$nreads,mito$mtfrac,labels=mito$RatID)
abline(h=0.1,v=6E6,col="red")

plot(mito$nreads,mito$mtfrac,pch=19,col="lightblue",xlab="No. reads", ylab="mtDNA proportion",cex=3)
text(mito$nreads,mito$mtfrac,labels=mito$Group)
abline(h=0.1,v=6E6,col="red")

```

## Screen for mitochondrial enrichment

Mito fraction versus whole cell irrespective of training.

First exclude any mito fractions with mt reads less than 10%.

Also exclude mito fractions with less than 3m reads.

Instead of normalising by reads per million, we are normalising by reads per million (mito transcripts).

Then exclude any gene with detection less than 10 reads per sample across all samples.
This way we can be sure the genes are real.

This cuts the number of genes down from 20k to 6.9k.

```{r,normalise}

mito <- subset(mito,nreads>3e6 & mtfrac > 0.1)
ssm <- ss[which(ss$RatID %in% mito$RatID),]
xm <- x[,which(colnames(x) %in% rownames(ssm))]
xmito <- rbind(xm[grep("Mt-",rownames(xm) ),],xm[grep("Mt_",rownames(xm) ),])
xmnorm <- xm/colSums(xmito)*1e6

xmnorm <- xmnorm[which(apply(xm,1,min)>10),]
dim(xmnorm)

```

Now to calculate the mito enrichment factor.

This is defined as the normalised score (mito) / normalised score (whole cell).

This is calculated for each rat and a median enrichment factor is determined.

As anticipated, the list of top candidates includes mitochondrial transcripts, but also others such as protein coding and 
snRNAs

```{r,enrichmentfactor}

r1 <- xmnorm$`33`/xmnorm$`1`
r15 <- xmnorm$`47`/xmnorm$`15`
r2 <- xmnorm$`34`/xmnorm$`2`
r26 <- xmnorm$`58`/xmnorm$`26`
r27 <- xmnorm$`59`/xmnorm$`27`
r28 <- xmnorm$`60`/xmnorm$`28`
r32 <- xmnorm$`64`/xmnorm$`32`

xd <- data.frame(r1,r15,r2,r26,r27,r28,r32,row.names=rownames(xmnorm) )
xd$median <- apply(xd,1,median)

xd2 <- merge(xd,xmnorm,by=0)
rownames(xd2) <- xd2[,1] ; xd2[,1] = NULL
xd2 <- xd2[order(-xd2$median),]

head(xd2,30) %>% kbl() %>% kable_paper("hover", full_width = F)

# mito genes
rbind(xd2[grep("Mt-",rownames(xd2) ),],xd2[grep("Mt_",rownames(xd2) ),]) %>% kbl() %>% kable_paper("hover", full_width = F)

```

## Investigate top candidates

To dig deeper into this set of genes, I ran a heatmap.
It shows that there is some enrichment as one might expect.

I also show the barpl

```{r,candidates}

candidates <- rownames(head(xd2,30))
candidates_df <- xmnorm[which(rownames(xmnorm) %in% candidates),]
heatmap.2(as.matrix(candidates_df),trace="none",scale="row",margin=c(5,15),Colv="none",dendrogram="row")

sapply(1:nrow(candidates_df),function(i) {
  wc <- as.numeric(candidates_df[i,1:7])
  mit <- as.numeric(candidates_df[i,8:14])
  wtest <- wilcox.test(wc,mit,paired=TRUE)
  HEAD = paste("paired Wilcox p =",signif(wtest$p.value,2))
  nam <- rownames(candidates_df)[i]
  dat <- list("whole cell"=wc,"mito"=mit)
  boxplot(dat,ylab="normalised expression",cex=0,col="white",main=nam)
  beeswarm(dat,add=TRUE,cex=1.5,pch=19)
  mtext(HEAD)
  vec <- as.numeric(candidates_df[i,])
  barplot(as.numeric(vec),names.arg=colnames(candidates_df),main=nam ,ylab="normalised expression")
  mtext(HEAD)
} )

```

## Session info

```{r}

sessionInfo()

```
