---
title: "Transcriptome sequencing of mitochondrial and whole cell fraction of the C2C12 cell line"
author: "Mark Ziemann"
date: "21/04/2020"
output:
    rmarkdown::html_document:
        toc: true
theme: cosmo
---

## Methods
Three RNA samples underwent RNA-seq analysis. Library #1 used Qiagen RNeasy columns; #2 used Zymo
DirectZol columns). Library #3 was prepared from RNA extracted from isolated mitochondria from the
same tissue.
Libraries were generated by the Deakin Genomics Facility and sequenced on the MiSeq system generating
paired end 150 bp reads.
Reads underwent adapter and quality (q>=20) trimming with 
[Skewer](https://www.ncbi.nlm.nih.gov/pubmed/24925680) version 0.2.2.

Read 1: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Read 2: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

Reads were then were mapped to the rat genome (Ensembl version 99) with 
[STAR aligner](https://www.ncbi.nlm.nih.gov/pubmed/23104886) version 2.7.3a.
I was having issues with a low number of mapped reads, despite BLAST searches showing that the
sequences were good, and I did a bit of troubleshooting. In the end I found that read 2 doesn't 
Map very well, which could be related to something about the adapter sequence of the lit that was 
used. To get around this, I only used read 1 in this analysis.
I also ran MultiQC to summarise the QC checks that were done.
STAR generated counts were imported into R for analysis.

## Read counts
Import the STAR counts. We can also include some info out of the Ensembl GTF file including gene name
and gene class.
```{r}
# import the 3 column table
tmp<-read.table("3col.tsv",header=F)
library("reshape2")
# convert the 3 col table into a standard count matrix 
x<-as.matrix(acast(tmp, V2~V1, value.var="V3"))
# tidy up the column headers
colnames(x)<-sapply(strsplit(colnames(x),"_"),"[[",1)
#dont forget gene names
g<-read.table("../ref/Rattus_norvegicus.Rnor_6.0.99.genenames.tsv",row.names=1)
x<-merge(g,x,by=0)
# gene accession numbers also have gene symbol and gene biotype
rownames(x)=paste(x$Row.names,x$V2,x$V3,sep="_")
# get rid of the columns that aren't required anymore
x$Row.names=x$V2=x$V3=NULL
# show the top 6 lines of data
head(x)
```

## Normalise and look at the top expressed genes
```{r}
# library size normalisation
xx <- x/colMeans(x) * 1000000 
head(xx)
# top expressed genes in library 1
head(xx[order(-xx$`1`),])
# top expressed genes in library 2
head(xx[order(-xx$`2`),])
# top expressed genes in library 3
head(xx[order(-xx$`3`),])
```

## Find out the RNAs enriched for mito localisation

```{r}
# exclude genes with no counts in the mito sample
xx<-xx[which(xx$`3`>0),]
# calculate ratio of mito to whole cell
xx$e1<-xx$`3`/xx$`1`
xx$e2<-xx$`3`/xx$`2`
# write tsv file for future reference
write.table(xx,file="rna.tsv",sep="\t",quote=FALSE)
# Top 50 enriched in lib 3 as compared to 1
kable(head(xx[order(-xx$e1),],50),caption="Table 1. Top 50 enriched in lib 3 as compared to 1",format="pandoc")
# Top 50 enriched in lib 3 as compared to 2
kable(head(xx[order(-xx$e2),],50),caption="Table 2. Top 50 enriched in lib 3 as compared to 2",format="pandoc")
```

## Now focus on lincRNAs
```{r}
xxx<-xx[grep("lincRNA",rownames(xx)),]
# enriched in lib 3 as compared to 1
kable(head(xxx[order(-xxx$e1),],50),caption="Table 3. Enriched in lib 3 as compared to 1",format="pandoc")
# Enriched in lib 3 as compared to 2
kable(head(xxx[order(-xxx$e2),],50),caption="Table 4. Enriched in lib 3 as compared to 2",format="pandoc")
```

## Session info
```{r}
sessionInfo()
```
