---
title: "Transcriptome sequencing of whole cell and mito fractions of sedentary and trained rats"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
    rmarkdown::html_document:
        toc: true
theme: cosmo
---

## Methods

Rats were kept in sedentary conditions or were trained. 
RNA was isolated from whole cell and mito fractions.
Libraries were generated by the Deakin Genomics Facility and sequenced on the MiSeq system generating
paired end 150 bp reads.
Reads underwent adapter and quality (q>=20) trimming with 
[Skewer](https://www.ncbi.nlm.nih.gov/pubmed/24925680) version 0.2.2.

Read 1: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Read 2: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

Reads were then were mapped to the rat genome (Ensembl version 99) with 
[STAR aligner](https://www.ncbi.nlm.nih.gov/pubmed/23104886) version 2.7.3a.
I was having issues with a low number of mapped reads, despite BLAST searches showing that the
sequences were good, and I did a bit of troubleshooting. In the end I found that read 2 doesn't 
Map very well, which could be related to something about the adapter sequence of the lit that was 
used. To get around this, I only used read 1 in this analysis.
I also ran MultiQC to summarise the QC checks that were done.
STAR generated counts were imported into R for analysis.

## Read counts

Import the STAR counts. We can also include some info out of the Ensembl GTF file including gene name
and gene class.

```{r,begin}

# libraries
library("reshape2")
library("DESeq2")
library("mitch")
library("gplots")

# import the 3 column table
tmp<-read.table("3col.tsv",header=F)

# convert the 3 col table into a standard count matrix 
x<-as.matrix(acast(tmp, V2~V1, value.var="V3"))
# tidy up the column headers
colnames(x)<-sapply(strsplit(colnames(x),"_"),"[[",1)
#dont forget gene names
g<-read.table("../ref/Rattus_norvegicus.Rnor_6.0.99.genenames.tsv",row.names=1)
x<-merge(g,x,by=0)
# remove underscores from the gene biotype because it might cause issues later
x$V3<-gsub("_","",x$V3)
# gene accession numbers also have gene symbol and gene biotype
rownames(x)=paste(x$Row.names,x$V2,x$V3,sep="_")
# get rid of the columns that aren't required anymore
x$Row.names=x$V2=x$V3=NULL
# show the top 6 lines of data
head(x)

```

## Samplesheet

```{r,samplesheet}

samplesheet <- read.table("samplesheet.tsv",header=TRUE)
samplesheet$UDI <- gsub("UDI_0","",samplesheet$UDI)
samplesheet$UDI <- gsub("UDI_","",samplesheet$UDI)
samplesheet

```

## Overall clustering with multidimensional scaling

Sample 95 is a HUMAN mito control sample.
Sample 96 is a RAT mito control sample.
These should not be inluded in the main results.

```{r,mds1}

mds <- cmdscale(dist(t(x)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds , xlab="Coordinate 1", ylab="Coordinate 2",
  type = "n" , xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(x) )

x <- x[,which(! colnames(x) %in% c("95","96"))]

mds <- cmdscale(dist(t(x)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds , xlab="Coordinate 1", ylab="Coordinate 2",
  type = "n" , xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(x) )

par(mar=c(5,10,5,3))
barplot(colSums(x),horiz=TRUE,las=2,main="number of reads per sample",cex.names=0.5)


```


## Functions

```{r,func}

run_de <- function(ss,xx){

y <- round(xx)

# MDS
mds <- cmdscale(dist(t(y)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds , xlab="Coordinate 1", ylab="Coordinate 2",
  type = "n" , xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(y) )

# DE
dds <- DESeqDataSetFromMatrix(countData=y, colData = ss, design = ~ trt)
dds <- DESeq(dds)
de <- DESeq2::results(dds)
de <- de[order(de$pvalue),]
up <- rownames(subset(de, log2FoldChange>0 & padj<0.05 ))
dn <- rownames(subset(de, log2FoldChange<0 & padj<0.05 ))
str(up)
str(dn)

# MA plot
sig <-subset(de, padj < 0.05 )
GENESUP <- length(up)
GENESDN <- length(dn)
SUBHEADER = paste(GENESUP, "up, ", GENESDN, "down")
ns <-subset(de, padj > 0.05 )
plot(log2(de$baseMean),de$log2FoldChange,
     xlab="log2 basemean", ylab="log2 foldchange",
     pch=19, cex=0.5, col="dark gray",
     main="smear plot")
points(log2(sig$baseMean),sig$log2FoldChange,
       pch=19, cex=0.5, col="red")
mtext(SUBHEADER)

# heatmap
yn <- y/colSums(y)*1000000
yf <- yn[which(rownames(yn) %in% rownames(de)[1:50]),]
mycols <- gsub("0","yellow",ss$trt)
mycols <- gsub("1","orange",mycols)
colfunc <- colorRampPalette(c("blue", "white", "red"))
heatmap.2(  as.matrix(yf), col=colfunc(25),scale="row",
    ColSideColors =mycols ,trace="none",
    margin = c(10,15), cexRow=0.6, cexCol=0.8 , main="Top 50 genes by p-val")
mtext("yellow=ctrl, orange=trt")

return(de)
}

```

## Set up contrasts

1. Whole tissue S versus T

2. Mito fraction S versus T

3. Whole versus mito fraction in S

4. Whole versus mito fraction in T

```{r,split}

ss1 <- subset(samplesheet,fraction=="wholetissue")
ss1$trt <- grepl("T",ss1$Group)*1
x1 <- x[,which(colnames(x) %in% rownames(ss1))]

ss2 <- subset(samplesheet,fraction=="mito")
ss2$trt <- grepl("T",ss2$Group)*1
x2 <- x[,which(colnames(x) %in% rownames(ss2))]

ss3 <- subset(samplesheet,Group=="S")
ss3$trt <- grepl("mito",ss3$fraction)*1
x3 <- x[,which(colnames(x) %in% rownames(ss3))]

ss4 <- subset(samplesheet,Group=="T")
ss4$trt <- grepl("mito",ss4$fraction)*1
x4 <- x[,which(colnames(x) %in% rownames(ss4))]

```

## DE

Here, were using DESeq2 to perform differential expression analysis for the specified
contrasts.
The run_de function does the analysis and generate the charts.
Here we actually run the analysis.

```{r,de,fig.height=8,fig.width=8}

ss1
de1 <- run_de(ss1,x1)
head(de1,20)
write.table(de1,file="de1.tsv",quote=FALSE,sep="\t")

ss2
de2 <- run_de(ss2,x2)
head(de2,20)
write.table(de2,file="de2.tsv",quote=FALSE,sep="\t")

ss3
de3 <- run_de(ss3,x3)
head(de3,20)
write.table(de3,file="de3.tsv",quote=FALSE,sep="\t")

ss4
de4 <- run_de(ss4,x4)
head(de4,20)
write.table(de4,file="de4.tsv",quote=FALSE,sep="\t")

```

## Session info

```{r}

sessionInfo()

```
