---
title: "Transcriptome sequencing of whole cell and mito fractions of sedentary and trained rats"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
    rmarkdown::html_document:
        toc: true
theme: cosmo
---

## Methods

Rats were kept in sedentary conditions or were trained. 
RNA was isolated from whole cell and mito fractions.
Libraries were generated by the Deakin Genomics Facility and sequenced on the NovaSeq system.

Fastqc and MultiQC were run to summarise the QC checks that were done.

Reads were then were mapped to the rat genome (Ensembl version 99) with Kallisto then
imported to R for analysis with DESeq2.
Pathway level analysis was then done using mitch with Reactome gene sets.

## Read counts

Import the Kallisto transcript counts.
We can also include some info out of the Ensembl GTF file including gene name
and gene class.

```{r,begin}

# libraries
library("reshape2")
library("DESeq2")
library("mitch")
library("gplots")

# import the 3 column table
tmp<-read.table("3col.tsv.gz",header=F)

# convert the 3 col table into a standard count matrix 
x<-as.matrix(acast(tmp, V2~V1, value.var="V3"))
# tidy up the column headers
colnames(x)<-sapply(strsplit(colnames(x),"_"),"[[",1)
head(x)

#dont forget gene names
g<-read.table("../ref/Rattus_norvegicus.Rnor_6.0.cdna+ncrna.gene_names.tsv",row.names=1)
g$gene_ID <- paste(g$V2,g$V3,g$V4)
head(g)
g[,1:3]=NULL
x<-merge(g,x,by=0)
rownames(x) <- x[,1]
x[,1]=NULL
# aggregate Tx data to genes
xx <- aggregate(. ~ gene_ID,x,sum)
# now round to integers so that DESeq2 doesn't fail
rownames(xx) <- xx[,1]
xx[,1]=NULL
x <- round(xx)
head(x)

```

## Samplesheet

```{r,samplesheet}

samplesheet <- read.table("samplesheet.tsv",header=TRUE)
samplesheet$UDI <- gsub("UDI_0","",samplesheet$UDI)
samplesheet$UDI <- gsub("UDI_","",samplesheet$UDI)
samplesheet$label <- paste(samplesheet$fraction,samplesheet$Group)
samplesheet

```

## Overall clustering with multidimensional scaling

Sample 95 is a HUMAN mito control sample.
Sample 96 is a RAT mito control sample.
These should be excluded from main results.

```{r,mds1,fig.height=8,fig.width=8}

ss <- samplesheet
ss <- ss[order(ss$UDI),]
colours = c('pink', 'lightblue','lightgreen','gray')
mds <- cmdscale(dist(t(x)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds*1.05 , cex=2, pch=19, xlab="Coordinate 1", ylab="Coordinate 2",
  col = colours[as.factor(ss$label)] ,  type = "p" ,
  xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(x) )
legend('topright', col=colours, legend=levels(as.factor(ss$label)), pch = 16, cex = 1)

x <- x[,which(! colnames(x) %in% c("95","96"))]

colours = c('pink', 'lightblue','lightgreen','gray')
mds <- cmdscale(dist(t(x)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds*1.05 , cex=2, pch=19, xlab="Coordinate 1", ylab="Coordinate 2",
  col = colours[as.factor(ss$label)] ,  type = "p" ,
  xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(x) )
legend('topright', col=colours, legend=levels(as.factor(ss$label)), pch = 16, cex = 1)

```

```{r,nreads,fig.height=8,fig.width=8}

par(mar=c(5,10,5,3))
barplot(colSums(x),horiz=TRUE,las=2,main="number of reads per sample",cex.names=0.5)
par(mai=c(1.02,0.82,0.82,0.42))

```

## Check purity of mito fraction samples

Here I'm quantifying the mitochondrial read fraction.
That is the number of mt reads divided by the total number of reads.
We can see that purity is highly variable.

```{r,mito_purity,fig.height=8,fig.width=8}

par(mar=c(5,10,5,3))
mtfrac <-  colSums(x[grep("^MT",rownames(x)),]) / colSums(x) 
barplot(mtfrac,horiz=TRUE,las=2,main="Proportion mitochondrial reads",cex.names=1)
par(mai=c(1.02,0.82,0.82,0.42))

mylevels <- levels(as.factor(ss$label))
mylevels
y <- x[,which(ss$label==mylevels[1])]
mitoS <- colSums(y[grep("^MT",rownames(y)),]) / colSums(y)
y <- x[,which(ss$label==mylevels[2])] 
mitoT <- colSums(y[grep("^MT",rownames(y)),]) / colSums(y)
y <- x[,which(ss$label==mylevels[3])] 
wholeS <- colSums(y[grep("^MT",rownames(y)),]) / colSums(y)
y <- x[,which(ss$label==mylevels[4])] 
wholeT <- colSums(y[grep("^MT",rownames(y)),]) / colSums(y)

boxplot(mitoS,mitoT,wholeS,wholeT,names=mylevels,ylab="mito frac")

median(wholeS)
median(wholeT)

table(mitoS>0.4)
table(mitoT>0.4)

ss$mtfrac <- mtfrac

```


## Functions

```{r,func}

run_de <- function(ss,xx){

y <- round(xx)

# MDS
colours = c('yellow', 'orange')
mds <- cmdscale(dist(t(y)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds*1.05 , cex=2 , pch=19, xlab="Coordinate 1", ylab="Coordinate 2",
  col = colours[as.factor(ss$trt)] ,  type = "p" , 
  xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(y) )
legend('topright', col=colours, legend=c("ctrl","trt"), pch = 16, cex = 1.5)

# DE
dds <- DESeqDataSetFromMatrix(countData=y, colData = ss, design = ~ trt)
dds <- DESeq(dds)
de <- DESeq2::results(dds)
de <- de[order(de$pvalue),]
up <- rownames(subset(de, log2FoldChange>0 & padj<0.05 ))
dn <- rownames(subset(de, log2FoldChange<0 & padj<0.05 ))
str(up)
str(dn)

# MA plot
sig <-subset(de, padj < 0.05 )
GENESUP <- length(up)
GENESDN <- length(dn)
SUBHEADER = paste(GENESUP, "up, ", GENESDN, "down")
ns <-subset(de, padj > 0.05 )
plot(log2(de$baseMean),de$log2FoldChange,
     xlab="log2 basemean", ylab="log2 foldchange",
     pch=19, cex=0.5, col="dark gray",
     main="smear plot")
points(log2(sig$baseMean),sig$log2FoldChange,
       pch=19, cex=0.5, col="red")
mtext(SUBHEADER)

# heatmap
yn <- y/colSums(y)*1000000
yf <- yn[which(rownames(yn) %in% rownames(de)[1:50]),]
mycols <- gsub("0","yellow",ss$trt)
mycols <- gsub("1","orange",mycols)
colfunc <- colorRampPalette(c("blue", "white", "red"))
heatmap.2(  as.matrix(yf), col=colfunc(25),scale="row",
    ColSideColors =mycols ,trace="none",
    margin = c(10,15), cexRow=0.6, cexCol=0.8 , main="Top 50 genes by p-val")
mtext("yellow=ctrl, orange=trt")

return(de)
}

```

## Set up contrasts

1. Whole tissue S versus T

2. Mito fraction S versus T

3. Whole versus mito fraction in S

4. Whole versus mito fraction in T

```{r,split}

ss1 <- subset(samplesheet,fraction=="wholetissue")
ss1$trt <- grepl("T",ss1$Group)*1
x1 <- x[,which(colnames(x) %in% rownames(ss1))]

ss2 <- subset(samplesheet,fraction=="mito")
ss2$trt <- grepl("T",ss2$Group)*1
x2 <- x[,which(colnames(x) %in% rownames(ss2))]

ss3 <- subset(samplesheet,Group=="S")
ss3$trt <- grepl("mito",ss3$fraction)*1
x3 <- x[,which(colnames(x) %in% rownames(ss3))]

ss4 <- subset(samplesheet,Group=="T")
ss4$trt <- grepl("mito",ss4$fraction)*1
x4 <- x[,which(colnames(x) %in% rownames(ss4))]

```

## DE

Here, were using DESeq2 to perform differential expression analysis for the specified
contrasts.
The run_de function does the analysis and generate the charts.
Here we actually run the analysis.

1. Whole tissue S versus T: 1 DEG

2. Mito fraction S versus T: 0 DEGs

3. Whole versus mito fraction in S: 22754 DEGs

4. Whole versus mito fraction in T: 22406 DEGs

These results suggest that:

* The differences between S and T are very subtle.

* Intra-group variation is fairly large.

```{r,de,fig.height=8,fig.width=8}

ss1
de1 <- run_de(ss1,x1)
as.data.frame(de1[1:20,])
write.table(de1,file="de1.tsv",quote=FALSE,sep="\t")

ss2
de2 <- run_de(ss2,x2)
as.data.frame(de1[1:20,])
write.table(de2,file="de2.tsv",quote=FALSE,sep="\t")

ss3
de3 <- run_de(ss3,x3)
head(de3,20)
write.table(de3,file="de3.tsv",quote=FALSE,sep="\t")

ss4
de4 <- run_de(ss4,x4)
head(de4,20)
write.table(de4,file="de4.tsv",quote=FALSE,sep="\t")

```

## Second pass DE analysis

Here we are going to exclude the mito samples with < 40% mito reads.
Contrast 1 is not repeated.

2. Mito fraction S versus T: 0 DEGs

3. Whole versus mito fraction in S: 20 DEGs

4. Whole versus mito fraction in T: 0 DEGs


```{r, de2, fig.height=8,fig.width=8}

ss0 <- subset(ss, fraction=="mito" & mtfrac>0.4)
ss <- subset(ss, fraction!="mito")
ss <- rbind(ss,ss0)

ss2 <- subset(ss,fraction=="mito")
ss2$trt <- grepl("T",ss2$Group)*1
x2 <- x[,which(colnames(x) %in% rownames(ss2))]

ss3 <- subset(ss,Group=="S")
ss3$trt <- grepl("mito",ss3$fraction)*1
x3 <- x[,which(colnames(x) %in% rownames(ss3))]

ss4 <- subset(ss,Group=="T")
ss4$trt <- grepl("mito",ss4$fraction)*1
x4 <- x[,which(colnames(x) %in% rownames(ss4))]

ss2
de2 <- run_de(ss2,x2)
as.data.frame(de2[1:20,])
write.table(de2,file="de2b.tsv",quote=FALSE,sep="\t")

ss3
de3 <- run_de(ss3,x3)
head(de3,20)
as.data.frame(de3[1:20,])
write.table(de3,file="de3b.tsv",quote=FALSE,sep="\t")

ss4
de4 <- run_de(ss4,x4)
head(de4,20)
as.data.frame(de4[1:20,])
write.table(de4,file="de4b.tsv",quote=FALSE,sep="\t")

```

## Pathway analysis with mitch

Here we are doing a gene set analysis with my R package called mitch.
I'm using gene sets downloaded from Reactome 5th Dec 2020.

We lost 46% of genes after converting from rat to human.

There were 259 differentially regulated gene sets (FDR<0.05).
Of these 81 were downregulated and 178 were upregulated.


```{r, mitch1, fig.height=8,fig.width=8}

#download.file("https://reactome.org/download/current/ReactomePathways.gmt.zip",
#    destfile="ReactomePathways.gmt.zip")
#unzip("ReactomePathways.gmt.zip")
genesets<-gmt_import("ReactomePathways.gmt")

# i need to get some data from biomart to link rat and human gene IDs

mart <- read.table("mart_export.txt",header=TRUE,sep="\t")
gt <- mart[,c("Gene.stable.ID","Human.gene.name")]

rownames(de1) <- sapply( strsplit(rownames(de1)," ") , "[[", 2)
rownames(de1) <- sapply( strsplit(rownames(de1),"\\.") , "[[", 1)

m <- mitch_import(as.data.frame(de1),"DESeq2",geneTable=gt)

res<-mitch_calc(m,genesets,priority="significance")

nrow(subset(res$enrichment_result,p.adjustANOVA<0.05))

head(res$enrichment_result,20)

mitch_barplot <- function(res){
  sig <- head(subset(res$enrichment_result,p.adjustANOVA<0.05),30)
  sig <- sig[order(sig$s.dist),]
  par(mar=c(3,25,1,1)); barplot(sig$s.dist,horiz=TRUE,las=2,cex.names = 0.6,cex.axis = 0.6,
    names.arg=sig$set,main="Enrichment score") ;grid()
}

mitch_barplot(res)

nrow(subset(res$enrichment_result,p.adjustANOVA<0.05&s.dist<0))
nrow(subset(res$enrichment_result,p.adjustANOVA<0.05&s.dist>0))

unlink("skeletal_mitch1.html")
mitch_report(res,outfile="skeletal_mitch1.html")

```

## Session info

```{r}

sessionInfo()

```
